src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -533,6 +534,7 @@ private BoundExpression VisitIndexPatternIndexerAccess(BoundImplicitIndexerAcces
             {
                 if (isLeftOfAssignment && indexerAccess.GetRefKind() == RefKind.None)
                 {
+                    AddPlaceholderReplacement(argumentPlaceholder, integerArgument);
```

Comment:
>AddPlaceholderReplacement(argumentPlaceholder, integerArgument); [](http://example.com/codeflow?start=20&length=64)

What feels strange or confusing about this method now, is the fact that we do not respect ```cacheAllArgumentsOnly``` parameter on this code path. Two consumers make an extra effort to ensure that arguments are actually cached if we hit this code path. Before, this method didn't have to deal with caching arguments at all. I am not saying, that we absolutely have to clean this up in context of this PR, but this feels like something we should follow up on.